clang/lib/Driver/ToolChains/Arch/RISCV.cpp

```diff
@@ -194,14 +197,7 @@ void riscv::getRISCVTargetFeatures(const Driver &D, const llvm::Triple &Triple,
   if (Arg *A = Args.getLastArg(options::OPT_mabi_EQ)) {
     bool IsPureCapability = isCheriPurecapABIName(A->getValue());
     if (IsPureCapability) {
-      auto ISAInfo = llvm::RISCVISAInfo::parseFeatures(
-          Triple.isArch32Bit() ? 32 : 64,
-          std::vector<std::string>(Features.begin(), Features.end()));
-      if (!ISAInfo) {
-        handleAllErrors(ISAInfo.takeError(), [&](llvm::StringError &ErrMsg) {
-          D.Diag(diag::err_invalid_feature_combination) << ErrMsg.getMessage();
-        });
-      } else if (!(*ISAInfo)->hasExtension("xcheri")) {
+      if (!ISAInfo->hasExtension("xcheri")) {
```

Comment:
This is not quite the same, the first ISAInfo compuation just looks at the arch string and genenerates a features list. This one also factors in the various -mfoo flags from handleTargetFeaturesGroup().

I think we might actually need both calls?

Reply:
I *think* right now we don't handle any flags that could cause issues there but I think I noticed it with my WIP changes to add a std-compat mode.

Reply:
handleTargetFeaturesGroup should never be doing anything with the ISA though, that should stay the same. It's for various non-ISA features like save-restore.

Reply:
Fair point, right now nothing after parsing the arch string has any meaningful effect. So happy to keep as is.